api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -337,9 +337,13 @@ private void discontinueExistingOrdersIfNecessary(Order order, Boolean isRetrosp
 			return;
 		}
 
+		Date asOfDate = null;
+		if (isRetrospective)
+			asOfDate = order.getDateActivated();
+
 		//Mark first order found corresponding to this DC order as discontinued.
 		List<? extends Order> orders = getActiveOrders(order.getPatient(), order.getOrderType(), order.getCareSetting(),
-		    null);
+		    asOfDate);
```

Comment:
On the design call, we agreed to what you're saying so you can ignore my comment, there is more details on the ticket, see Darius' last comment or the updated ticket description under dev notes